reposerver/repository/repository.go

```diff
@@ -1046,9 +1046,13 @@ func GetAppSourceType(ctx context.Context, source *v1alpha1.ApplicationSource, p
 		return "", err
 	}
 	if appSourceType != nil {
+		if !discovery.IsManifestGenerationEnabled(*appSourceType, enableGenerateManifests) {
+			log.Debugf("Manifest generation is disabled for '%s'. Assuming plain YAML manifest.", *appSourceType)
+			return v1alpha1.ApplicationSourceTypeDirectory, nil
```

Comment:
Hm. Just realized. If we get to this point, and `v1alpha1.ApplicationSourceTypeDirectory` is disabled, should we throw an error? Or should we just return directory type and rely on the caller to properly handle the configuration (e.g. prevent manifest generation)?

Reply:
The `v1alpha1.ApplicationSourceTypeDirectory` is a special case because we use it for both YAMLs and jsonnet. So we need to return Directory type here and directory type manifest generator should double-check if manifest generation is enabled: https://github.com/argoproj/argo-cd/blob/40149cdae75cc3ec3465aaccd05381c8147e2357/reposerver/repository/repository.go#L1148